---
title: "manhattan"
---

## Single GWAS
```{r}
library(topr)
```

```{r, echo=FALSE}
mh.width=10
mh.height=4
mh.legend.height=4.5
```



```{r, fig.width=mh.width, fig.height=mh.height}
manhattan(CD_UKBB)
```

Annotate the top association peaks with the nearest gene

```{r, fig.width=mh.width, fig.height=mh.height}
manhattan(CD_UKBB, annotate=5e-09)
```

Extract the lead snps and display them in red on the manhattan plot

```{r, fig.width=mh.width, fig.height=mh.legend.height, warning=F, message=F}
lead_snps <- get_lead_snps(CD_UKBB, thresh = 5e-09, region_size = 10000000)
```

```{r, fig.width=mh.width, fig.height=mh.legend.height, warning=F, message=F}
manhattan(list(CD_UKBB, lead_snps), 
          annotate=c(5e-100, 5e-09), 
          color=c("darkblue","red"), 
          legend_labels=c("CD UKBB", "Lead snp's"), verbose=F) 
```

Change the annotation to vertical and italic. Reduce the size of all text by setting the <code>scale</code> argument to 0.9. Move the legend labels to the top.

```{r, fig.width=mh.width, fig.height=mh.legend.height, warning=F, message=F}

manhattan(list(CD_UKBB, lead_snps), 
          annotate=c(5e-100, 5e-09), 
          color=c("darkblue","red"), 
          legend_labels=c("CD UKBB", "Lead snp's"), 
          size=c(1,2),
          shape=c(19,8), 
          angle = 90, 
          label_fontface = "italic", 
          nudge_y = 5, 
          scale=0.9, 
          legend_position = "top", 
          ymax=32,  
          even_no_chr_lightness = c(0.8,0.5),
          verbose=F) 
```


```{r, fig.width=mh.width, fig.height=mh.legend.height, warning=F, message=F}
top_snps_finngen <- get_lead_snps(CD_FINNGEN)
manhattan(list(CD_UKBB, lead_snps, top_snps_finngen),  
          color=c("red","darkred","gray40"),
          label_fontface = c("italic","bold","plain"), 
          angle=c(90,0,0),
          legend_labels = c("CD","top snps UKB", "top snps Finngen"),
          annotate=1e-20)
```


## Multi GWAS

## Highlight known and novel loci


